cmd/codegen/generator/generator.go

```diff
@@ -57,6 +57,11 @@ type Config struct {
 	// Dev indicates that the codegen should use the local SDK instead of the published one.
 	// This is only relevant when ClientOnly is true.
 	Dev bool
+
+	// DependenciesRef is the list of all dependencies used by the module.
+	// This is used for client generator to automatically serves the dependencies
+	// on connection.
+	DependenciesRef []string
```

Comment:
Hmmm. I also think we're missing something here.

What if the user has renamed the dependency? e.g. I've got a module `github.com/shykes/hello`, but set it's `name` field to `"foobar"`. It should be served as `"foobar"`, I think this information is getting lost.

I think we actually need the *entire* module config struct. It's not just the `name`, it's the `pin`, and any other config we might choose to add later on.

Reply:
`asString` already uses the pin and the repo, but I need to change the logic to also include the name.
I might pass a flag as argument then instead of all references as flags